### DISCUSSION TITLE
Hiding certain .ISO/.IMG files depending of which boot mode is used Legacy or UEFI

### DISCUSSION THREAD

**Discussion by mikolaj612** [ventoy/Ventoy#2068] [General]
Hello.
As thread name states - is it possible to hide certain files when using UEFI and/or Legacy?

I do have some stuff that I use with CSM mode and some will work only with UEFI and it would be great to hide/assign things that work with certain settings to avoid issues and to create shorter (clearer) menu list.

Anyone got some ideas about it?

**Comment by steve6375** [comment]
https://www.ventoy.net/en/plugin_imagelist.html

**Reply by mikolaj612** [reply]
> https://www.ventoy.net/en/plugin_imagelist.html

Fantastic! 
What if I've got files that are compatible with UEFI and LEGACY mode as well and I wan't to include them in both scenarios?
I don't want to duplicate some heavy ISO files on my USB stick - how to overcome this?

**Reply by steve6375** [reply]
Just specify the file in both legacy whitelist and uefi64 whitelist

**Reply by mikolaj612** [reply]
> Just specify the file in both legacy whitelist and uefi64 whitelist

When I do whitelist my menu list is empty.

I've got folder called ISO which I want to share across CSM and UEFI mode.
And one called CSM which should be used only for LEGACY and second one called UEFI for newer PC's.

For now I've got CSM and UEFI stuff listed properly depending on which BIOS mode is used, however I can't get my CSM/UEFI compatible iso files located at ISO to be listed in both mode - those are WINPE and Ubuntu installer.

**Reply by steve6375** [reply]
Use /iso/CSM folder structure?

**Reply by steve6375** [reply]
Otherwise use same folders and just whitelist all wanted files
